File: apps/otel-python/otelpython/api/root.py
# SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
# SPDX-License-Identifier: Apache-2.0

import logging

import fastapi

from otelpython.telemetry import metrics

logger = logging.getLogger(__name__)

router = fastapi.APIRouter()


@router.get("/", tags=["root"])
async def root_handler() -> str:
    logger.info("[handler] Version")
    metrics.request_counter.add(1, {"target_service": "root"})
    return "OpenTelemetry Lab / Python"

File: apps/otel-js/src/routes/root.js
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

const { getLogger } = require("../telemetry/shared");

async function rootRoutes(fastify) {
  fastify.get("/", async (request, reply) => {
    const logger = getLogger();
    logger.emit({
      severityText: "info",
      body: "Root handler",
    });
    return reply.send("OpenTelemetry Lab / Javascript");
  });
}

module.exports = rootRoutes;
